atom/renderer/api/atom_api_web_frame.cc

```diff
@@ -142,7 +164,12 @@ void WebFrame::SetSpellCheckProvider(mate::Arguments* args,
   spell_check_client_.reset(new SpellCheckClient(
       language, auto_spell_correct_turned_on, args->isolate(), provider));
   web_frame_->View()->SetSpellCheckClient(spell_check_client_.get());
-  web_frame_->SetTextCheckClient(spell_check_client_.get());
+  // Set spellchecker for all live frames in the same process or
+  // in the sandbox mode for all live sub frames to this WebFrame.
+  FrameSpellChecker spell_checker(
+      spell_check_client_.get(),
+      content::RenderFrame::FromWebFrame(web_frame_));
+  content::RenderFrame::ForEach(&spell_checker);
```

Comment:
The steps look good but there might be a small timing issue here: in subsequent calls, the webframes' TextCheckClient will be dangling pointers inbetween the calls to
`spell_check_client_.reset()` and
`ForEach() ->render_frame->GetWebFrame()->SetTextCheckClient()`

Seems like we'd want to keep the previous spell_check_client_ alive in temporary unique_ptr until after `ForEach`

Reply:
Followup question, is it certain that the `ForEach` will change all the frames that hold a pointer to the old spellchecker?

E.g. is it possible for `GetRenderView()`, `GetMainRenderFrame()`, or `IsMainFrame()` to change so that subsequent calls to `ForEach` might change a different set of frames than the last call?

Reply:
> there might be a small timing issue here:

👍 , in that case would it be better to initialize `SpellCheckClient` once per frame and just reset the states on subsequent calls ? I am not sure about the reason for current implementation. @zcbenz what do you think about this ?

> is it certain that the ForEach will change all the frames that hold a pointer to the old spellchecker?

Yup, `ForEach` is certain to iterate over all live frames in the process. The only reason for the main frame checks are to make sure that in sandbox mode frames with similar site instances in different windows only get assigned to the spellchecker their parent frame is assigned with.

Reply:
You can keep the new pointer and then do a swap:

```c++
std::unique_ptr client(new SpellCheckClient());
ForEach()->Set(client.get());
spell_check_client_.swap(client);
```

Reply:
Sorry for the delay, have addressed the changes requested. Thanks!